### PR TITLE
Connect with spotify API

### PR DESCRIPTION
**Pull request by hubmelco**
Connected to the spotify API to allow users to search for tracks, albums, and artists. The idea is to allow them to search for these things so they can populate their posts with the data

### CODE REVIEW

File: src/repository/songDAO.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const json = await response.json();
+
+    if (json.error) {
+        const {status} = json.error;
+        if (status !== 401) {
+            throw json.error;
+        }
+        // 401 means token expired so retry method after getting a new token
+        token = await getToken();
+        return await getSongs(q, type, offset, retries + 1);
```
Comment (kylefwalters) [inline src/repository/songDAO.js:41]: `type` here is not defined.

File: src/services/songService.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    try {
+        const {tracks} = await songDAO.getSongs(q, offset);
+        const filtered = cleanData(tracks.items);
+    
+        const total = tracks.total - offset;
+        let previous = "";
+        if (offset > 0) {
+            const previousOffset = (offset - (total - (total - tracks.limit)) > 0) ? offset - (total - (total - tracks.limit)) : 0;
+            previous = constructPageURL(query, previousOffset, type);
```
Comment (kylefwalters) [inline src/services/songService.js:38]: Also here

File: src/services/songService.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        }
+    });
+}
+
+/**
+ *  Creates a url that will jump between paginated results based on the offset
+ * 
+ * @param query The input parameters given by the user
+ * @param offset The offset to start from
+ * @param type The type of query (artist, track, album)
```
Comment (kylefwalters) [inline src/services/songService.js:81]: Parameter `type` is here, but not defined in `constructPageURL`.

File: src/services/songService.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        return "";
+    }
+    // Maybe put the base url in a .env variable
+    let url = `http://localhost:3000/songs?type=track`;
+    for (const key in query) {
+        if (query[key] !== undefined) {
+            url += `&${key}=${query[key]}`;
+        }
+    }
+    url+=`offset=${offset}`;
```
Comment (kylefwalters) [inline src/services/songService.js:95]: You forgot to add a `&` at the beginning, resulting in an incorrect URL.

**Review by kylefwalters — COMMENTED** [review]
Looks good, just a few bug fixes and should be good to go.

File: src/repository/songDAO.js (hunk 1 above)
Comment (hubmelco) [reply src/repository/songDAO.js:41]: Good catches! I removed the type and I missed these. Thanks

**Pull request merged**